fairlearn/grid_search/binary_protected_attribute/binary_classification.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import copy
+import numpy as np
+
+from . import utilities
+
+
+def _weight_function(y_val, a_val, L, p_ratio, a0_val):
+    if a_val == a0_val:
+        return 2 * y_val - 1 - L * p_ratio
+    else:
+        return 2 * y_val - 1 + L
+
+
+def _generate_weights(y, protected_attribute, L, p_ratio, a0_val):
+    weight_func = np.vectorize(_weight_function)
+    return weight_func(y, protected_attribute, L, p_ratio, a0_val)
+
+
+def demographic_parity(learner,
```

Comment:
This name seems a little bit less descriptive to me than it probably should be. I was expecting something like grid_search_for_demographic_parity, because DP is only the metric

Reply:
Same for BGL

Reply:
I was thinking that all the rest of that (not to mention binary_protected_attribute) was implicit from the path

Reply:
But people don't use it that way. In the code you'd have
```
demographic_parity(...)
```
so as a user I'm very confused about what that will even return. Apparently it returns models, which is funny because the function is named after a metric which doesn't indicate that there will be models.

Reply:
I could prefix with `sweep_` if you feel that would be better (and I can see that making a lot of sense). But I don't like `grid_search_for_demographic_parity` because
- This is already in a `grid_search` module
- If we're not going to have the namespace be the description, we also need to add something about binary protected attributes as well

Reply:
Definitely better. I guess the problem I'm seeing is that people don't read imports, they read code, so nobody will know what your module is unless they are completely confused and then look at your imports. We don't necessarily want them to do that. If they really care about the details they can always do that, but it shouldn't be that the code is unreadable. Adding a verb like sweep would solve that.

Reply:
Change made and pushed. Adding the verb definitely distinguishes it from just the metrics named.